src/nav/navActions.js

```diff
@@ -29,89 +32,83 @@ export const resetToMainTabs = (): GenericNavigationAction =>
  */
 
 /** Only call this via `doNarrow`.  See there for details. */
-export const navigateToChat = (narrow: Narrow): NavigationAction =>
-  StackActions.push({ routeName: 'chat', params: { narrow } });
+export const navigateToChat = (narrow: Narrow): GenericNavigationAction =>
+  StackActions.push('chat', { narrow });
 
-export const navigateToUsersScreen = (): NavigationAction =>
-  StackActions.push({ routeName: 'users' });
+export const navigateToUsersScreen = (): GenericNavigationAction => StackActions.push('users');
```

Comment:
They do note (as they do on their "navigating without the navigation
    prop" doc [3]) that it's best to use the `navigation` object when
    one can:
    
    """
    It's highly recommended to use the methods on the navigation object
    instead of using action creators and `dispatch`. It should only be
    used for advanced use cases.
    """

Interesting. Seems like migrating to that would be a good followup commit, then (not necessarily within this PR.)

I think probably the bulk of these `navActions` actions are only ever invoked from within a React component. Perhaps most of them even from a React component that already has a `navigation` prop, and there's `useNavigation` for a convenient way to handle some of the others. So that might let us eliminate most of these.

And when we do, it looks like we'd get proper type-checking! When I try `this.props.navigation.navigate()` in a random component and start entering arguments, it looks like I get Flow telling me whether I have a proper route name and then whether I have the right arguments for that route. That's pretty nifty.

Reply:
> And when we do, it looks like we'd get proper type-checking! When I try `this.props.navigation.navigate()` in a random component and start entering arguments, it looks like I get Flow telling me whether I have a proper route name and then whether I have the right arguments for that route. That's pretty nifty.

Nice! Happy to do that as a followup.